### PR TITLE
[KEYCLOAK-7154] KCINIT and console mode docs

### PR DESCRIPTION
**Pull request by patriot1burke**
kcinit docs and how to write authenticator/required action support for console mode.

### CODE REVIEW

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:5]: Command line interface with what?
Configure themselves for what?

Perhaps something like this, with you adjusting my words for accuracy:
Many applications need a command line interface with an identity provider for the application to administer and configure itself.

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:14]: ...you will...
Contractions are generally not preferred in tech docs. I won't point out others, but it is a good practice not to use them in this style of writing.

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:19]: The second disadvantage is that if there are other CLI apps using {project_name}, you will have to login multiple times, once for each CLI app that is secured by {project_name} as there is no SSO available on the command line.

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:42]: Changing this to active voice with a non-conditional verb phrasing...

After entering this example, `kcinit` prompts you to login using a text-based interface. Then, `kcinit` performs the OAuth protocol behind the scenes to obtain a token, and outputs the token to STDOUT.

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:56]: Instead of "you will have to" use the stronger phrase "you must".

...create a dedicated client...

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:80]: "up front"?

I think the sentence is better without those two words.

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:93]: Remove "you can" and just tell them to log in (two words in verb form).

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:103]: ...command outputs the...

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:111]: It then checks to see if...

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:117]: ...single sign-on...

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:118]: Split into a new paragraph, starting with "The first way..." to parallel the next paragraph that starts with "The second way..."

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [inline securing_apps/topics/cli-apps/cli-apps.adoc:119]: Change "have to be" to "must"

File: server_development/topics/auth-spi.adoc
Comment (matthewhelmke) [inline server_development/topics/auth-spi.adoc:809]: mix-in, unless "mixin" appears in the openID docs as written.

**Review by matthewhelmke — CHANGES_REQUESTED** [review]
You are already a pretty good writer, @patriot1burke. Rather than just take what you have and edit it for you completely, which I am willing to do, I have instead given recommended edits with an explanation. I think I caught and noted everything, but this gives an example to help future contributions start out even better. Let me know if you would rather I just edit for you without the (hopefully not pedantic, but rather helpful) commentary and I will.

**Review by pedroigor — APPROVED** [review]
Content-wise, it looks good for me.

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (patriot1burke) [reply securing_apps/topics/cli-apps/cli-apps.adoc:14]: I think I remember O'Reilly telling me that a afew times years ago :)

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (patriot1burke) [reply securing_apps/topics/cli-apps/cli-apps.adoc:5]: Maybe just:

"Many applications have a command line interface."

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (patriot1burke) [reply securing_apps/topics/cli-apps/cli-apps.adoc:19]: nice.

**Comment by patriot1burke** [comment]
@matthewhelmke Ok, I'll add your suggestions and recommit...

FYI, don't merge as I need another PR approved that sets up the download for this tool.

File: securing_apps/topics/cli-apps/cli-apps.adoc
Comment (matthewhelmke) [reply securing_apps/topics/cli-apps/cli-apps.adoc:5]: That works and is simpler. I like it.

**Comment by matthewhelmke** [comment]
Okay. I'll watch for the new commit and will wait for word from you before merging.

**Pull request closed without merging**
